sources/packages/api/src/route-controllers/institution/institution.aest.controller.ts

```diff
@@ -239,15 +246,85 @@ export class InstitutionAESTController extends BaseController {
    * @param payload authorizations to be associated with the user.
    * @returns Primary identifier of the created resource.
    */
-  // TODO: Add API Responses and validations. This method will be worked in the upcoming PR.
+  @ApiNotFoundResponse({ description: "Institution not found." })
+  @ApiUnprocessableEntityResponse({
+    description:
+      "User to be added was not found on BCeID Account Service " +
+      "or the user does not belong to the same institution " +
+      "or the user already exists " +
+      "or a second legal signing authority is trying to be set when one is already in place.",
+  })
   @Post(":institutionId/user")
   async createInstitutionUserWithAuth(
-    institutionId: number,
+    @Param("institutionId", ParseIntPipe) institutionId: number,
     @Body() payload: CreateInstitutionUserAPIInDTO,
   ): Promise<PrimaryIdentifierAPIOutDTO> {
     return this.institutionControllerService.createInstitutionUserWithAuth(
       institutionId,
       payload,
     );
   }
+
+  /**
+   * Update the user authorizations for the institution user.
+   * @param userName user to have the permissions updated.
+   * @param payload permissions to be updated.
+   */
+  @ApiNotFoundResponse({
+    description: "User to be updated not found.",
+  })
+  @ApiUnprocessableEntityResponse({
+    description:
+      "The user is not active" +
+      " or the user permission is being updated in a way that no admin will be present" +
+      " or a second legal signing authority is trying to be set and only one is allowed.",
+  })
+  @Patch("user/:userName")
+  async updateInstitutionUserWithAuth(
+    @Param("userName") userName: string,
+    @Body() payload: UpdateInstitutionUserAPIInDTO,
+  ): Promise<void> {
+    await this.institutionControllerService.updateInstitutionUserWithAuth(
+      userName,
+      payload,
+    );
+  }
+
+  /**
+   * Get institution user by user name(guid).
+   * @param userName user name (guid).
+   * @returns institution user details.
+   */
+  @ApiNotFoundResponse({
+    description: "User not found.",
+  })
+  @Get("user/:userName")
+  async getInstitutionUserByUserName(
+    @Param("userName") userName: string,
+  ): Promise<InstitutionUserAPIOutDTO> {
+    return this.institutionControllerService.getInstitutionUserByUserName(
+      userName,
+    );
+  }
+
+  /**
+   * Update the active status of the user.
+   * @param userName unique name of the user to be updated.
+   * @param payload information to enable or disable the user.
+   */
+  @ApiNotFoundResponse({
+    description: "User to be updated not found.",
+  })
+  @Patch("user-status/:userName")
```

Comment:
`user/:userName/status` ?

Reply:
The endpoint was already there but I will adjust it on the next PR 😉